src/Core/Checkout/Cart/Calculator.php

```diff
@@ -25,23 +22,16 @@ class Calculator
     private $priceCalculator;
 
     /**
-     * @var PercentagePriceCalculator
+     * @var PriceCalculatorInterface[]
      */
-    private $percentagePriceCalculator;
-
-    /**
-     * @var AbsolutePriceCalculator
-     */
-    private $absolutePriceCalculator;
+    private $calculators;
 
     public function __construct(
         QuantityPriceCalculator $priceCalculator,
-        PercentagePriceCalculator $percentagePriceCalculator,
-        AbsolutePriceCalculator $absolutePriceCalculator
+        iterable $calculators
```

Comment:
`$calculators` must be annotated in phpdoc so the Static Code Analysis can reason about the constructor:

```
    /**
     * @param PriceCalculatorInterface[] $calculators
     */
```